game-app/game-core/src/main/java/games/strategy/triplea/ui/BattlePanel.java

```diff
@@ -67,6 +64,7 @@ public final class BattlePanel extends ActionPanel {
   BattlePanel(final GameData data, final MapPanel map, final JFrame parent) {
     super(data, map);
     battleWindow = new JDialog(parent);
+    battleWindow.setDefaultCloseOperation(WindowConstants.DO_NOTHING_ON_CLOSE);
```

Comment:
Do you know readily what's the default close operation? Can you give some more update about why the default close operation is set here?

I think this update is potentially a bit more risky as we could have issues with the game not closing when the main window is closed due to hidden dialogs. I think we have had a few examples of that problem in the past (and it's pretty hard to track down).

Reply:
I see, we are setting this earlier than we did previously, so there is no change here.  :+1: